### PR TITLE
[Jormungandr] Pagination added in the response

### PR DESCRIPTION
**Pull request by kadhikari**
Pagination added in /freefloatings_nearby

Jira: https://jira.kisio.org/browse/NAVITIAII-3486

### CODE REVIEW

File: source/jormungandr/jormungandr/external_services/free_floating.py [hunk 1]
```diff
@@ -69,9 +69,30 @@ def response_marshaller(cls, response):
...
+        if resp.get('pagination') is not None:
+            items_on_page = resp.get('pagination').get('items_on_page', len(resp.get('free_floatings', [])))
+            items_per_page = resp.get('pagination').get('items_per_page', args.get('count'))
+            start_page = resp.get('pagination').get('start_page', args.get('start_page'))
+            total_result = resp.get('pagination').get('total_result', len(resp.get('free_floatings', [])))
+        else:
+            items_on_page = len(resp.get('free_floatings', []))
+            items_per_page = args.get('count')
+            start_page = args.get('start_page')
+            total_result = items_on_page
```
Comment (azime) [inline source/jormungandr/jormungandr/external_services/free_floating.py]:
```suggestion
            items_on_page = resp.get('pagination', {}).get('items_on_page', len(resp.get('free_floatings', [])))
            items_per_page = resp.get('pagination', {}).get('items_per_page', args.get('count'))
            start_page = resp.get('pagination', {}).get('start_page', args.get('start_page'))
            total_result = resp.get('pagination', {}).get('total_result', len(resp.get('free_floatings', [])))

```
No ?

File: source/jormungandr/jormungandr/interfaces/v1/serializer/free_floating.py [hunk 2]
```diff
@@ -78,9 +88,20 @@ class FreeFloatingSerializer(serpy.DictSerializer):
     distance = NestedPropertyField(attr='distance')
 
 
+class PaginationSerializer(serpy.DictSerializer):
+    start_page = NestedPropertyField(attr='start_page')
+    items_on_page = NestedPropertyField(attr='items_on_page')
+    items_per_page = NestedPropertyField(attr='items_per_page')
+    total_result = NestedPropertyField(attr='total_result')
+
+
```
Comment (azime) [inline source/jormungandr/jormungandr/interfaces/v1/serializer/free_floating.py:97]: Duplicate ([Here](https://github.com/CanalTP/navitia/blob/5866b150286046604cf2795b2d9fcdf388b9a80a/source/jormungandr/jormungandr/interfaces/v1/serializer/fields.py#L166)), no ?

Comment (kadhikari) [reply source/jormungandr/jormungandr/interfaces/v1/serializer/free_floating.py:97]: The function which exists already is for protobuf.

File: source/jormungandr/jormungandr/external_services/free_floating.py (hunk 1 above)
Comment (kadhikari) [reply source/jormungandr/jormungandr/external_services/free_floating.py]: :+1:

**Review by pbench — APPROVED** [review]

File: source/jormungandr/jormungandr/external_services/free_floating.py [hunk 3]
```diff
@@ -69,9 +69,27 @@ def response_marshaller(cls, response):
...
+        resp['pagination']['items_on_page'] = resp.get('pagination', {}).get(
+            'items_on_page', len(resp.get('free_floatings', []))
+        )
+        resp['pagination']['items_per_page'] = resp.get('pagination', {}).get(
+            'items_per_page', args.get('count')
+        )
+        resp['pagination']['start_page'] = resp.get('pagination', {}).get('start_page', args.get('start_page'))
+        resp['pagination']['total_result'] = resp.get('pagination', {}).get(
+            'total_result', len(resp.get('free_floatings', []))
+        )
```
Comment (AzHicham) [inline source/jormungandr/jormungandr/external_services/free_floating.py]: May be create `resp.get('pagination', {})` once instead of calling it 4 times ?

Comment (kadhikari) [reply source/jormungandr/jormungandr/external_services/free_floating.py]: :+1:

**Review by benoit-bst — APPROVED** [review]

**Pull request merged**